plugins/grid-bookmark/src/GridBookmarkWidget/components/BookmarkGrid.tsx

```diff
@@ -117,85 +97,58 @@ const BookmarkGrid = observer(function ({
                 : measureText('label'),
             editable: true,
           },
-          {
-            field: 'assemblyName',
-            headerName: 'Assembly',
-            width:
-              bookmarkRows.length > 0
-                ? measureGridWidth(bookmarkRows.map(row => row.assemblyName))
-                : measureText('assembly'),
-          },
+          ...(selectedAssembly === undefined
+            ? [
+                {
+                  field: 'assemblyName',
+                  headerName: 'Assembly',
+                  width:
+                    bookmarkRows.length > 0
+                      ? measureGridWidth(
+                          bookmarkRows.map(row => row.assemblyName),
+                        )
+                      : measureText('assembly'),
+                },
+              ]
+            : []),
         ]}
         onCellDoubleClick={({ row }) => {
           setDialogOpen(true)
           setDialogRow(row)
         }}
-        /* @ts-ignore */
         processRowUpdate={row => {
           const target = bookmarkRows[row.id]
           model.updateBookmarkLabel(target, row.label)
           return row
         }}
-        onProcessRowUpdateError={e => {
-          session.notify(e.message, 'error')
-        }}
+        onProcessRowUpdateError={e => session.notify(e.message, 'error')}
         checkboxSelection
         onRowSelectionModelChange={newRowSelectionModel => {
-          const selectedBookmarks = [] as IExtendedLabeledRegionModel[]
-          newRowSelectionModel.forEach((value: GridRowId) => {
-            selectedBookmarks.push({ ...bookmarkRows[value as number] })
-          })
-          model.setSelectedBookmarks(selectedBookmarks)
+          model.setSelectedBookmarks(
+            newRowSelectionModel.map(value => ({
```

Comment:
use typescript inference for the callback parameter, and also use map instead of forEach+push